app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -196,6 +196,8 @@ case class EvaluationResults(
     def loop(expr: Expr): ExpressionResult = expr match {
       case ctx @ FormCtx(_) =>
         evalDateExpr(recData, this)(DateFormCtxVar(ctx))
+      case Else(field1: Expr, field2: Expr) =>
+        loop(field1) orElse loop(field2)
```

Comment:
This looks weird, will this work for expression which mixes date and other types?

Like:

```
${startDate else 'You didnt select any date'}
```

Reply:
Thats a good point, haven't tested it. It will return Empty for non date values (when condition is met)

Reply:
Just added some changes. Basically if the date expression could not be evaluated i.e Empty, it tries to evaluate as a String as a fallback